mage_ai/cluster_manager/kubernetes/workload_manager.py

```diff
@@ -146,55 +151,64 @@ def create_deployment(
                 }
             )
 
-        if volume_host_path:
-            volumes.append(
-                {
-                    'name': 'mage-data',
-                    'hostPath': {
-                        'path': volume_host_path
-                    },
-                }
-            )
-        deployment_template_spec = dict()
+        stateful_set_template_spec = dict()
         if service_account_name:
-            deployment_template_spec['serviceAccountName'] = service_account_name
+            stateful_set_template_spec['serviceAccountName'] = service_account_name
 
-        deployment = {
+        stateful_set = {
             'apiVersion': 'apps/v1',
-            'kind': 'Deployment',
+            'kind': 'StatefulSet',
             'metadata': {
-                'name': deployment_name,
+                'name': name,
                 'labels': {
-                    'app': deployment_name
+                    'app': name
                 }
             },
             'spec': {
                 'selector': {
                     'matchLabels': {
-                        'app': deployment_name
+                        'app': name
                     }
                 },
                 'replicas': 1,
                 'minReadySeconds': 10,
                 'template': {
                     'metadata': {
                         'labels': {
-                            'app': deployment_name
+                            'app': name
                         }
                     },
                     'spec': {
                         'terminationGracePeriodSeconds': 10,
                         'containers': containers,
                         'volumes': volumes,
-                        **deployment_template_spec
+                        **stateful_set_template_spec
                     }
                 },
+                'volumeClaimTemplates': [
+                    {
+                        'metadata': {
+                            'name': 'mage-data'
+                        },
+                        'spec': {
+                            'accessModes': [
+                                'ReadWriteOnce'
```

Comment:
this could cause problem. we usually use ReadWriteMany. but we can make it configurable.
1G below might not always enough as well.

Reply:
ok i can make this configurable